src/components/TrackPicker.test.js

```diff
@@ -0,0 +1,196 @@
+import React from 'react';
+import { render, fireEvent, waitFor }  from '@testing-library/react';
+import {TrackPicker} from './TrackPicker';
+import '@testing-library/jest-dom'
+import config from "./../config.json";
+
+
+describe('TrackPicker', () => {
+    const tracks = {
+        1: config.defaultTrackProps,
+        2: config.defaultTrackProps,
+        3: config.defaultTrackProps,
+    }
+    const availableColors = ["greys", "ygreys", "reds", "plainColors", "lightColors"];
+    const availableTracks = [{"files": [{"name": "fileA1.vg", "type": "graph"},
+                                        {"name": "fileA2.gbwt", "type": "haplotype"}]},
+                             {"files": [{"name": "fileB1.gbwt", "type": "haplotype"},
+                                        {"name": "fileB2.gam", "type": "read"}]},
+                             {"files": [{"name": "fileC1.xg", "type": "graph"}]}];
+
+    it('should render without errors', async () => {
+        const fakeOnChange = jest.fn();
+        const { queryByTestId } = render(
+            <TrackPicker
+                tracks={tracks}
+                availableTracks={availableTracks}
+                availableColors={availableColors}
+                onChange={fakeOnChange}
+            />
+        );
+
+        const trackPickerButton = queryByTestId("TrackPickerButton");
+
+        expect(trackPickerButton).toBeTruthy();
+
+        fireEvent.click(trackPickerButton);
+
+        expect(queryByTestId("file-type-select-component1")).toBeTruthy();
+
+        expect(queryByTestId("file-type-select-component2")).toBeTruthy();
+
+        expect(queryByTestId("file-select-component1")).toBeTruthy();
+
+        expect(queryByTestId("settings-button-component1")).toBeTruthy();
+
+        expect(queryByTestId("delete-button-component1")).toBeTruthy();
+
+        expect(queryByTestId("track-add-button-component")).toBeTruthy();
+    });
+
+
+    it('should add track items when the add button is pressed', async () => {
```

Comment:
It looks like this test and some of the others might be more duplicative than they need to be with the TrackPickerDisplay tests. We should be able to assume that the TrackPickerDisplay works already, and so e.g. clicking its add button will add a track. What we really care abut here is making sure the popup opens and closes, and making sure changed actually propagate through when made in the TrackPickerDisplay, and that they do so at the right time relative to opening/closing the dialog.